lnurl/parse_url.js

```diff
@@ -0,0 +1,64 @@
+const {bech32} = require('bech32');
+
+const {decode} = bech32;
+const asLnurl = n => n.substring(n.startsWith('lightning:') ? 10 : 0);
+const bech32CharLimit = 2000;
+const emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+const prefix = 'lnurl';
+const sslProtocol = n => (!!/onion/.test(n) ? 'http://' : 'https://');
```

Comment:
I don't think the request function will work with HTTP, so onion wouldn't work here

Reply:
So just error out if it's onion saying it's not supported?

Reply:
Yeah to support onion I need to adjust my simple request function to support http requests, in nodejs they are separated into two libraries and I only did https so far